### PR TITLE
MLDB-1162 html plugin

### PR DESCRIPTION
**Pull request by jeremybarnes**
This is a hangover from the pre-open sourced MLDB version.

Not to be merged until there is a clear DOD.

FYI @nicolaskruchten

### CODE REVIEW

**Comment by nicolaskruchten** [comment]
So you want me to spec out some HTML parsing functions?

**Comment by guyd** [comment]
Can we split the commits of the external libraries and those of the plugins so that we can review the relevant code more easily.

**Comment by jeremybarnes** [comment]
Commits have been split into external and internal.

File: html/html_function.h [hunk 1]
```diff
@@ -0,0 +1,48 @@
+/** tokensplit.h                                      -*- C++ -*-
```
Comment (guyd) [inline html/html_function.h]: Wrong header

File: html/html_function.h [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    Function to parse strings for tokens and insert separators
+*/
+
+#pragma once
+
+#include "mldb/server/function.h"
+#include "mldb/types/value_description_fwd.h"
+#include "mldb/types/optional.h"
+
+namespace Datacratic {
```
Comment (guyd) [inline html/html_function.h]: We should remove that namespace.

**Comment by FinchPowers** [comment]
This PR is big enough to choke my browser. The part I saw had no test coverage of the new parser. Is that correct?

**Pull request merged**